server/src/main/kotlin/org/elaastic/player/dashboard/SequenceMonitoringModel.kt

```diff
@@ -46,26 +55,23 @@ class SequenceMonitoringModel(
      */
     fun setLearners(newLearnersList: MutableList<LearnerMonitoringModel>) {
         learners.clear()
-
-        val newLearnersListSorted: MutableList<LearnerMonitoringModel> = when (this.executionContext) {
-            ExecutionContext.FaceToFace -> this.sortWithFaceToFaceBehavior(newLearnersList)
-            else -> this.sortWithBlendedOrRemoteBehavior(newLearnersList)
-        }
-
-        learners.addAll(newLearnersListSorted)
+        learners.addAll(
+            when (this.executionContext) {
+                ExecutionContext.FaceToFace -> this.sortWithFaceToFaceBehavior(newLearnersList)
+                else -> this.sortWithBlendedOrRemoteBehavior(newLearnersList)
+            }
+        )
     }
 
     /**
      * Sort the learner when the sequence is in FaceToFace execution context
      *
-     * If the Phase 1 is active, then we want the learner still writing their
-     * answer at the top.
+     * If the Phase 1 is active, then we want the learner still writing their answer at the top.
      *
-     * If the Phase 2 is active, then we want the learner still evaluating at
-     * first and more than the learner who didn't answer the question.
+     * If the Phase 2 is active, then we want the learner still evaluating at first and more than the learner who didn't
+     * answer the question.
      *
-     * If the Phase 3 is active, then we want the learner who didn't answer and
-     * evaluate at first.
+     * If the Phase 3 is active, then we want the learner who didn't answer and evaluate at first.
      */
     private fun sortWithFaceToFaceBehavior(
```

Comment:
Won't it be possible to name the sort function to express the sort strategy ?

Beside that, I add a comment on the sorting implementation.
The same list is sorted two times in `sortWithBlendedOrRemoteBehavior`. I guess it is an error. It not the equivalent to sort a list on 2 criteria.

Reply:
It's not easy to describe the sorting algorithm in the name of the function, the sorting strategy for a `FaceToFace` sequence depend on the active phase.

I don't understand your comment about the `sortWithBlendedOrRemoteBehavior` function.

Reply:
For the 1st point : I thought their may be a natural way to describe what we want to achieve thought the sorting but if it is not the case, no problem, than the naming seems to be the best we can do.

About my 2nd comment, I take an illustrative scenario : 
```
someList.sortBy { it.name }
someList.sortBy { it.birthDate }
```
 
I've investigated more deeply now, and I've discoverded that this usage of `sortBy` is correct.
It is correct because this sort function acts by mutation and is stable. It means that equal elements preserve their order relative to each other after sorting.

My 1st concerns was that the 2nd sorting overrides the 1st one. Because it is stable, it is not the case.
One can do a 2nd comment : there are methods more efficient doing the sort according to the 2 criteria simultaneously. But we don't care of such a performance here because there isn't a lot of data.

Last thing : I'm a fan of functional programming so I would have avoided mutation here.
With a functional approach it would have looked like : 
```
someList.sortedWith(compareBy<Person> { it.name}.thenBy { it.birthDate})
``` 

A lot of talkings just to say I've learn today about Kotlin and sorting.
I'm definitely sure I prefer the functional way over the imperative way.
Nevertheless this code is correct and not need to be changed.

Thanks for reading :)

Reply:
1. OK

2. Here is the difference between the imperative and the functional way :

Imperative
```kotlin
// Sorted By name alphabetically
newLearnersList.sortBy { it.learnerName }
// Sort by number of states in progress descending
newLearnersList.sortByDescending { it.getLevelByStateCell(LearnerMonitoringModel.StateCell.IN_PROGRESS) }
```

Functional
```kotlin
newLearnersList.sortWith(
    // Sorted By name alphabetically
    compareBy<LearnerMonitoringModel> { it.learnerName }
        // Sort by number of states in progress descending
        .thenByDescending { it.getLevelByStateCell(LearnerMonitoringModel.StateCell.IN_PROGRESS) }
)
```
I found the functional less easy to read. And with imperative, is just like an English sentence. 
So for now, as you suggest, I stay with the actual implementation.

Reply:
Those good properties are not critical here. 
For me it became the natural way of thinking, so I tend use it spontaneously.

Reply:
Just to be complete, it can be written this way : 
```kotlin
newLearnersList.sortWith(compareBy<LearnerMonitoringModel>(
    { it.learnerName },
    { -it.getLevelByStateCell(LearnerMonitoringModel.StateCell.IN_PROGRESS) }
))
```